planb/create_cluster.py

```diff
@@ -686,13 +694,121 @@ def create_cluster(options: dict):
         # Undo stack sounds like a natural choice.
         #
         for region, sg in security_groups.items():
-            ec2 = boto3.client('ec2', region)
+            ec2 = ec2_client(region)
             info('Cleaning up security group: {}'.format(sg['GroupId']))
             ec2.delete_security_group(GroupId=sg['GroupId'])
 
         if options['use_dmz']:
             for region, ips in node_ips.items():
-                ec2 = boto3.client('ec2', region)
+                ec2 = ec2_client(region)
+                for ip in ips:
+                    info('Releasing IP address: {}'.format(ip['PublicIp']))
+                    ec2.release_address(AllocationId=ip['AllocationId'])
+
+        raise
+
+
+def extend_cluster(options: dict):
+    options['regions'] = [options['region']]
+
+    ec2 = ec2_client(options['region'])
+    running_instances = [
+        i
+        for i in list_instances(ec2, options['cluster_name'])
+        if i['State']['Name'] == 'running'
+    ]
+    if not running_instances:
+        msg = "Could not find any running EC2 instances for {} in {}".format(
+            options['cluster_name'],
+            options['region']
+        )
+        raise click.UsageError(msg)
+
+    # TODO: don't override docker image?
+    options = validate_artifact_version(options)
+    options = read_environment(options)
+
+    # List of IP addresses by region
+    node_ips = collections.defaultdict(list)
+
+    try:
+        # TODO: get it from a running instance details?
+        taupage_amis = find_taupage_amis(options['regions'])
+
+        subnets = get_subnets(
+            'dmz-' if options['use_dmz'] else 'internal-',
+            options['regions']
+        )
+        allocate_ip_addresses(
+            subnets, options['cluster_size'], node_ips,
+            take_elastic_ips=options['use_dmz']
+        )
+
+        if options['sns_topic'] or options['sns_email']:
+            alarm_topics = setup_sns_topics_for_alarm(
+                options['regions'],
+                options['sns_topic'],
+                options['sns_email']
+            )
+        else:
+            alarm_topics = {}
+
+        if options.get('hosted_zone'):
+            setup_dns_records(
+                options['cluster_name'],
+                options['hosted_zone'],
+                node_ips
+            )
+
+        cluster_sg = find_security_group_by_name(ec2, options['cluster_name'])
+        security_groups = {
+            options['region']: cluster_sg
+        }
+
+        # We should have up to 3 seeds nodes per DC
```

Comment:
Actually, it depends if the nodes in DC for dc_suffix exists or not. We want to be able to extend an existing DC with dc_suffix as well.